include/bactria/bactria.hpp

```diff
@@ -14,10 +14,47 @@
  */
 
 #ifndef BACTRIA_BACTRIA_HPP
-#define BACTRIA_BACTRIA_HPP
+#   define BACTRIA_BACTRIA_HPP
 
-#include <bactria/event.hpp>
-#include <bactria/phase.hpp>
-#include <bactria/region.hpp>
+#   include <bactria/Event.hpp>
+#   include <bactria/Plugin.hpp>
+#   include <bactria/Range.hpp>
+#   include <bactria/Phase.hpp>
+#   include <bactria/Sector.hpp>
+
+#   include <memory>
+
+namespace bactria
+{
+    struct Context
+    {
+        Context(plugin::plugin_handle_t h) noexcept : handle{h}
+        {}
+
+        Context(Context const&) noexcept = default;
+        auto operator=(Context const&) noexcept -> Context& = default;
+        Context(Context&&) noexcept = default;
+        auto operator=(Context&&) noexcept -> Context& = default;
+        ~Context() = default;
+
+        plugin::plugin_handle_t handle;
+    };
+
+    auto initialize() -> std::unique_ptr<Context>
+    {
+        return std::make_unique<Context>(plugin::load_plugin());
+    }
```

Comment:
Why the `unique_ptr`? Can you not just return a `Context`?

Reply:
I do not know yet if the Context may change in the future (in a way preventing Return Value Optimization), thus I'd rather return a pointer.

Reply:
Every major compiler implementes RVO and NRVO for a long time. RVO is even required since C++17. I think you can count on that ;)